### ISSUE TITLE
NVDA reads clickable when event handlers are attached to the root.

### ISSUE THREAD

**Issue by DarioR01** [w3c/aria#1684]
### Introductory information:
Since the introduction of React 17, React applications have started to delegate event handlers at the root level. Previously React used to delegate event handlers directly to the document node. _(More info [here](https://blog.saeloun.com/2021/07/08/react-17-event-delagation.html))._

Delegating events at the root level causes loss in semantics for assistive technology, and this makes them behave not as intended in specific cases involving event delegation.

### **Steps to reproduce:**

1. Open the below example in Firefox
2. Using NVDA, navigate the <img> element via down arrow keystroke
3. Notice how NVDA announces the graphic element as clickable

Take the following minimal code as an example:

```
<!DOCTYPE html>
<html lang='en'>

<head>
    <title>Basic HTML5 document</title>
    <meta charset='utf-8'>
</head>

<body>
    <div id="root">
            <img src="https://picsum.photos/200/300" alt="test text"/>
    </div>

    <script>
        // event delegation
        document.querySelector('#root').addEventListener('click', () => { console.log('click') })
    </script>
</body>

</html>
```

But if we now delegate the event to the document node and we follow the same steps to reproduce the bug described above:

```
<!DOCTYPE html>
<html lang='en'>

<head>
    <title>Basic HTML5 document</title>
    <meta charset='utf-8'>
</head>

<body>
    <div id="root">
            <img src="https://picsum.photos/200/300" alt="test text"/>
    </div>

    <script>
        // event delegation
        document.addEventListener('click', () => { console.log('click') })
    </script>
</body>

</html>
```
As you can hear, clickable is not being announced.

### Current behaviour
NVDA reads clickable in the first example provided. This creates a major issue when we don't want the event attached to the component to have any impact on user experience _(e.g. click tracking handlers)_. As mentioned at the beginning of this issue, all 17+ React versions are affected by this.

### Expected behaviour
We would expect NVDA not to read clickable in the first example provided.

### Possible Solution
A possible solution could be to create an aria attribute that allows authors to arbitrarily declare that an event is used for event delegation.

### Alternatives Solution
I am open to any other ideas that can solve this!!!

**Comment by WW3** [comment]
a. the problem lies in the bad decision to attach a click handler on a div
element (not keyboard accessible).
b. if the content should be served only to sighted mouse users, consider
adding aria-hidden on the div

Neil

**Comment by DarioR01** [comment]
We want our content to reach as many audiences as possible including keyboard-only users.
As mentioned this is an issue that React introduced. React was made aware of this last year as [this open issue](https://github.com/facebook/react/issues/20895) suggests. But they seem to not be interested in solving it.
Regardless of this, I already tried to re-open an issue with them just this morning: https://github.com/facebook/react/issues/23170

I also asked NVDA and they guided me here. A link to the NVDA issue can be found here: https://github.com/nvaccess/nvda/issues/13262

**Comment by eps1lon** [comment]
> But they seem to not be interested in solving it.

Because the technique they use was and still is relevant in web development. It's usually called "event delegation". 

I'm fine with escalating this issue to browsers if their a11y APIs expose these elements as clickable. 

But just claiming that event delegation isn't web standard just doesn't help. If it wouldn't be web standard then click events shouldn't bubble. But click events bubble do so web content authors need a way to tell AT that some click handlers are used for delegating and others aren't.

**Comment by DarioR01** [comment]
@eps1lon sorry if I misinterpreted your comment from the facebook/react issue. I have done more research to understand further. Now I believe that this bug should be fixed by NVDA or browser due to my example element not being focusable nor having interactive semantics. ([more here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics)).

I will see what NVDA have to say before escalating this to the browser, thank you for your time.

I will close this ticket for now.
